File: Interviews/GFG/random-problems/Hard/Flatten-binary-tree-to-linked-list/Flatten-binary-tree-to-linked-list.py
class Solution:
    def flatten(self, root):
        st = []
        st.append(root)
        
        while st:
            curNode = st.pop()
            if curNode.right:
                st.append(curNode.right)
            if curNode.left:
                st.append(curNode.left)
            
            if st:
                curNode.right = st[-1]
            curNode.left = None

File: Interviews/GFG/Daily-Problems/2022/October/Day-07/Flatten-binary-tree-to-linked-list.cpp
class Solution {
public:
    void flatten(Node *root) {
        vector<Node*> st;
        st.push_back(root);

        while (st.size()) {
            Node* curNode = st.back();
            st.pop_back();

            if (curNode->right) {
                st.push_back(curNode->right);
            }
            if (curNode->left) {
                st.push_back(curNode->left);
            }

            if (st.size()) {
                curNode->right = st.back();
            }
            curNode->left = NULL;
        }
    }
};